osu.Server.Spectator/Hubs/MultiplayerHub.cs

```diff
@@ -275,6 +276,30 @@ public async Task ChangeState(MultiplayerUserState newState)
             }
         }
 
+        public async Task ChangeBeatmapAvailability(BeatmapAvailability newBeatmapAvailability)
+        {
+            using (var userUsage = await GetOrCreateLocalUserState())
+            using (var roomUsage = await getLocalUserRoom(userUsage.Item))
+            {
+                var room = roomUsage.Item;
+
+                if (room == null)
+                    throw new InvalidStateException("Attempted to operate on a null room");
+
+                var user = room.Users.Find(u => u.UserID == CurrentContextUserId);
+
+                if (user == null)
+                    throw new InvalidStateException("Local user was not found in the expected room");
```

Comment:
This should also be an `InvalidOperationException`. This means there's a serious server-side error, not something triggered by the user.

Reply:
Other similar cases actually had this `InvalidStateException`, which caused me into thinking that it's fine using this exception, but I've forced them all to be `InvalidOperationException` while I was excluding coverage for the other cases (https://github.com/ppy/osu-server-spectator/pull/43)